pkg/apis/pubsub/v1alpha1/topic_types.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	duckv1beta1 "github.com/knative/pkg/apis/duck/v1beta1"
+	"github.com/knative/pkg/webhook"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Topic is a resource representing a Topic backed by Google Cloud Pub/Sub.
+type Topic struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Topic.
+	Spec TopicSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the Topic. This data may be out of
+	// date.
+	// +optional
+	Status TopicStatus `json:"status,omitempty"`
+}
+
+// Check that Topic can be validated, can be defaulted, and has immutable fields.
+var _ runtime.Object = (*Topic)(nil)
+var _ webhook.GenericCRD = (*Topic)(nil)
+
+// Check that Topic implements the Conditions duck type.
+var _ = duck.VerifyType(&Topic{}, &duckv1beta1.Conditions{})
+
+// TopicSpec defines an parameters for creating or publishing to a Cloud
+// Pub/Sub Topic depending the PropagationPolicy.
+type TopicSpec struct {
+	// Secret is the credential to be used to create and publish into the
+	// Cloud Pub/Sub Topic. The value of the secret entry must be a service
+	// account key in the JSON format
```

Comment:
Missing period. What should the key for this value be?

Reply:
That doc doesn't tell me what the content of the Secret should be. Should there be a single key with a JSON string value? If so what is the key? Should the JSON fields be elevated into the Secret as individual kv pairs?

It would be fine to reference some other doc that explains how the Secret should look, but the GCP doc referenced doesn't say anything about how to format a k8s Secret.